account_invoice_validate_send_email/models/account_invoice.py

```diff
@@ -69,14 +69,21 @@ def action_invoice_open(self):
             if invoice.type not in ("out_invoice", "out_refund"):
                 continue
             term = invoice.payment_term_id
-            orders = invoice.invoice_line_ids.mapped("sale_line_ids").mapped("order_id")
-            pickings = self.env["stock.picking"].search([("sale_id", "in", orders.ids)])
-
+            pickings = invoice.picking_ids
+            if not pickings:
+                orders = invoice.invoice_line_ids.mapped("sale_line_ids").mapped(
+                    "order_id"
+                )
+                pickings = self.env["stock.picking"].search(
+                    [("sale_id", "in", orders.ids)]
+                )
```

Comment:
```suggestion
                if orders:
                    pickings = self.env["stock.picking"].search(
                        [("sale_id", "in", orders.ids)]
                    )
```
Sorry I didn't catch this earlier.

Reply:
Thank you, I forgot too.